### PR TITLE
Issue #867 - UI for including screenshot in bug report

### PR DESCRIPTION
**Pull request by magsout**
Ok, so @miketaylr 

capture d ecran 2015-12-30 a 21 26 31

### CODE REVIEW

File: webcompat/templates/form.html [hunk 1]
```diff
@@ -37,19 +37,28 @@
...
-          {% endfor %}{% endif %}
-          {{ form.image(class_='ButtonUpload', accept='.jpe,.jpg,.jpeg,.gif,.png,.bmp') }}
+      <div class="r-Grid-cell r-Grid-cell--alignBottom r-all--1of2 r-maxS--2of2">
+        <div class="wc-Form-group  {% if form.image.errors %}wc-Form-error{% endif %}">
+          <label class="wc-Form-upload" for="image">
+            <div class="wc-Form-upload-wrapper">
+              <div class="wc-Form-upload-icon" aria-hidden="true">
+                <span class="wc-Icon wc-Icon--cloud-upload"></span>
+              </div>
+              {{ form.image.label(class_='wc-Form-label wc-Form-label--upload') }}
```
Comment (magsout) [inline webcompat/templates/form.html]: If you can remove the attribute `for`  for this `<label>`

Comment (miketaylr) [inline webcompat/templates/form.html]: I'll see if I can figure that out. ^_^

**Comment by miketaylr** [comment]
Whoa, speedy. :zap:  Will play with this in a bit.

File: webcompat/templates/form.html (hunk 1 above)
Comment (miketaylr) [inline webcompat/templates/form.html]: We might be able to just hand-write the `<label>` and get rid of the python part here (I think). Probably simplest.

**Comment by miketaylr** [comment]
This looks awesome @magsout. I'm gonna add an image preview via `FileReader` (if the browser supports it), otherwise it's hard to know if you've selected something. Once that's in place, I think we can land this and I'll extend it to work with the add-on screenshot stuff (and add drag & drop, etc.)

**Comment by miketaylr** [comment]
(that push was just to rebase against master)

**Comment by miketaylr** [comment]
We can improve the UX in future issues, a couple ideas:

1) ability to clear out file selection, right now you have to just select a new image
2) remove/modify "Add screenshot image" + icon
3) other stuff...

**Comment by miketaylr** [comment]
(it can look bad depending on the image, but that's OK I think for now... incremental improvements, etc.):

**Comment by magsout** [comment]
@miketaylr we can improve UX before merge if you want. Not a big deal.

**Comment by miketaylr** [comment]
@magsout if you have any suggestions, feel free to hack on them. I don't really have any amazing ideas, hence my "let's do it later". :p

When you get a chance, can you review the JS part?

**Comment by miketaylr** [comment]
I think I have one idea, let me try to come up with a quick mockup.

**Comment by miketaylr** [comment]
Here's one idea, stealing from the original design:

Once we have the upload "read" and the background-image set, we hide the label w/ icon and show a yellow bar on the bottom with a link "Click here to remove image" (I guess it would technically be a button, but I think styling as a link would be good). Once the user clicks that, we remove the bg-image, clear out the file input, and show the original "Upload screenshot image">

We could re-use that same bottom bar for the add-on use case where a user needs to check the checkbox (something like "[ ] Select to include screenshot image")

screen shot 2015-12-31 at 3 33 53 pm

What do you think?

**Comment by magsout** [comment]
wooot @miketaylr awesome :+1:

**Comment by magsout** [comment]
I added a small animation and replaced `<div>` by `<button>`.

I have one issue when I removed image uploaded:

capture d ecran 2016-01-05 a 12 35 06

**Comment by miketaylr** [comment]
Yay, thanks for cleaning up my markup/CSS ^_^. Let me try to reproduce that error, it's kinda wierd.

**Comment by miketaylr** [comment]
Ah, the weird view function error is because it's a button -- it's trying to submit or navigate or something. Will sprinkle some JS to fix it, then merge.

**Comment by miketaylr** [comment]
> Will sprinkle some JS to fix it

Or just use regular HTML `type=button`.

**Comment by magsout** [comment]
Ah yes, I writed `role` instead of type. Good catch @miketaylr

**Pull request merged**